instrumentation/trilogy/lib/opentelemetry/instrumentation/trilogy/patches/client.rb

```diff
@@ -0,0 +1,130 @@
+          }.freeze
+
+          MYSQL_COMPONENTS = %i[
+            single_quotes
+            double_quotes
+            numeric_literals
+            boolean_literals
+            hexadecimal_literals
+            comments
+            multi_line_comments
+          ].freeze
+
+          FULL_SQL_REGEXP = Regexp.union(MYSQL_COMPONENTS.map { |component| COMPONENTS_REGEX_MAP[component] })
+
+          def query(sql)
+            attributes = client_attributes
+
+            case config[:db_statement]
+            when :obfuscate
+              attributes[::OpenTelemetry::SemanticConventions::Trace::DB_STATEMENT] = obfuscate_sql(sql)
+            when :include
+              attributes[::OpenTelemetry::SemanticConventions::Trace::DB_STATEMENT] = sql
+            end
+
+            tracer.in_span(
+              database_span_name(sql),
+              attributes: attributes,
+              kind: :client
+            ) do
+              super(sql)
+            end
+          end
+
+          private
+
+          def obfuscate_sql(sql)
+            if sql.size > 2000
+              'SQL query too large to remove sensitive data ...'
+            else
+              obfuscated = sql.gsub(FULL_SQL_REGEXP, '?')
+              obfuscated = 'Failed to obfuscate SQL query - quote characters remained after obfuscation' if detect_unmatched_pairs(obfuscated)
+              obfuscated
+            end
+          end
+
+          def detect_unmatched_pairs(obfuscated)
+            # We use this to check whether the query contains any quote characters
+            # after obfuscation. If so, that's a good indication that the original
+            # query was malformed, and so our obfuscation can't reliably find
+            # literals. In such a case, we'll replace the entire query with a
+            # placeholder.
+            %r{'|"|\/\*|\*\/}.match(obfuscated)
+          end
+
+          def database_span_name(sql)
+            # Setting span name to the SQL query without obfuscation would
+            # result in PII + cardinality issues.
+            # First attempt to infer the statement type then fallback to
+            # current Otel approach {database.component_name}.{database_instance_name}
+            # https://github.com/open-telemetry/opentelemetry-python/blob/39fa078312e6f41c403aa8cad1868264011f7546/ext/opentelemetry-ext-dbapi/tests/test_dbapi_integration.py#L53
+            # This creates span names like mysql.default, mysql.replica, postgresql.staging etc.
+
+            statement_type = extract_statement_type(sql)
+
+            return statement_type unless statement_type.nil?
+
+            # fallback
+            'mysql'
+          end
+
+          def client_attributes
+            attributes = { ::OpenTelemetry::SemanticConventions::Trace::DB_SYSTEM => 'mysql' }
+            attributes[::OpenTelemetry::SemanticConventions::Trace::NET_PEER_NAME] = @connected_host if defined?(@connected_host)
```

Comment:
I peaked into https://github.com/github/trilogy/blob/9380a5844c844dd13ddd6dc761895898b17186a8/contrib/ruby/lib/trilogy.rb a bit and indeed it doesn't look like too much is available via public ruby methods for us to grab extra metadata at present. So it goes (_awesome gem tho, thank you to all the great GH folks for open sourcing it!_ 🎉 ). 

That being said all that's required according to the [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/3e380e249f60c3a5f68746f5e84d10195ba41a79/specification/trace/semantic_conventions/database.md#connection-level-attributes) (which itself is experimental so it leaves us wiggle room) is `db.system` and _either_ `net.peer.name` || `net.peer.ip`. So I think we're covered here. My only incredibly small bikeshedding thing is maybe we should set this to something like `unknown` in the event that `@connected_host` isn't defined, since downstream processors may assume that either `net.peer.name` or `net.peer.ip` will _always_ exist? Idk, not blocking at all, and given that your org is literally the only user of this instrumentation at the moment, I don't think it matters too much and defer to you on this.

Reply:
> db.system and either net.peer.name || net.peer.ip

TIL! Let me see what I can do. Perhaps I can use the original host provided to the connection and switch to the connected host when it is set.